java-squid/src/main/java/org/sonar/java/se/symbolicvalues/SymbolicValueRelation.java

```diff
@@ -174,9 +213,9 @@ private SymbolicValueRelation combineUnordered(SymbolicValueRelation relation) {
    */
   @CheckForNull
   protected Boolean implies(SymbolicValueRelation relation) {
-    if (v1.equals(relation.v1) && v2.equals(relation.v2)) {
+    if (leftOp.equals(relation.leftOp) && rightOp.equals(relation.rightOp)) {
       return relation.isImpliedBy(this);
-    } else if (v1.equals(relation.v2) && v2.equals(relation.v1)) {
+    } else if (leftOp.equals(relation.rightOp) && rightOp.equals(relation.leftOp)) {
       return relation.symmetric().isImpliedBy(this);
     }
     return null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Null is returned but a "Boolean" is expected. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2447)